server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/PdbDataManager.java

```diff
@@ -46,82 +56,149 @@
      * @return dataset from query
      * @throws ExternalDbUnavailableException
      */
-    public Dataset fetchRCSBEntry(final String pdbids) throws ExternalDbUnavailableException {
-
-        ParameterNameValue[] params = new ParameterNameValue[]{new ParameterNameValue("pdbids", pdbids),
-            new ParameterNameValue("format", "xml"),
-            new ParameterNameValue("customReportColumns", "structureTitle,expressionHost,classification,"
-                                                          + "compound,uniprotRecommendedName")};
-
-        String location = RCSB_SERVER + RCSB_TOOL + "?";
-
-        String rcsbpbdData = httpDataManager.fetchData(location, params);
-
+    public Dataset fetchRCSBEntry(final String pdbIds) throws ExternalDbUnavailableException {
         try {
-            JAXBContext jaxbContext = JAXBContext.newInstance("com.epam.catgenome.manager.externaldb.bindings.rcsbpbd");
-            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-
-            StringReader reader = new StringReader(rcsbpbdData);
-            Object uniprotObject = unmarshaller.unmarshal(reader);
-
-            Dataset fetchedDatasetData = null;
-            if (uniprotObject instanceof Dataset) {
-                fetchedDatasetData = (Dataset) uniprotObject;
-            }
-
-            return fetchedDatasetData;
-        } catch (JAXBException e) {
+            String query = replaceHttpSymbols(RCSB_ENTRY_QUERY, pdbIds);
+            return parseToDataset(QueryUtils.execute(rcsbApi.getDataset(query)), pdbIds);
+        } catch (RSCBResponseException e) {
             throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
-                    .ERROR_UNEXPECTED_FORMAT, location), e);
+                    .ERROR_UNEXPECTED_FORMAT, LOCATION), e);
         }
     }
 
-
     /**
-     *
      * @param pdbIds PDB id
      * @return data from query
      * @throws ExternalDbUnavailableException
      */
     public Dasalignment fetchPdbMapEntry(final String pdbIds) throws ExternalDbUnavailableException {
+        try {
+            String query = replaceHttpSymbols(PDB_MAP_ENTRY, pdbIds);
+            return parseToDasalignment(QueryUtils.execute(rcsbApi.getDasalignment(query)));
+
+        } catch (RSCBResponseException e) {
+            throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
+                    .ERROR_UNEXPECTED_FORMAT, LOCATION), e);
+        }
+    }
 
-        ParameterNameValue[] params = new ParameterNameValue[]{new ParameterNameValue("query", pdbIds)};
+    private String replaceHttpSymbols(String query, String pdbIds) {
+        return URLEncoder.DEFAULT.encode(String.format(query, pdbIds), StandardCharsets.UTF_8.toString());
 
-        String location = RCSB_SERVER + RCSB_PDP_MAP_TOOL + "?";
+    }
 
-        String rcsbpbdData = httpDataManager.fetchData(location, params);
+    private Dasalignment parseToDasalignment(DasalignmentDTO das) {
```

Comment:
Make parameter `final`. It is good conventions to have all method's parameters - `final`

Reply:
Fixed and understood!